### PR TITLE
Upgrade flow

### PR DESCRIPTION
**Pull request by TrySound**
Improved types of output function to reflect original type.

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -4,4 +4,7 @@
 # Example: // $FlowFixMe: This following line is borked because of reasons.
 # Example: // $ExpectError: This following line is correctly ignored by flow.
 suppress_comment= \\(.\\|\n\\)*\\$FlowFixMe
-suppress_comment= \\(.\\|\n\\)*\\$ExpectError
\ No newline at end of file
+suppress_comment= \\(.\\|\n\\)*\\$ExpectError
+
+[declarations]
+.*/node_modules/raf-stub/.*
```
Comment (alexreardon) [inline .flowconfig:10]: Can you explain this one for me?

File: src/index.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
 // @flow
 
-type WrapperFn = (...args: mixed[]) => void;
-type CancelFn = {|
+type WrapperFn<T> = {
+  [[call]]: (...T) => void,
```
Comment (alexreardon) [inline src/index.js:4]: I have not seen this before

**Review by alexreardon — APPROVED** [review]

File: .flowconfig (hunk 1 above)
Comment (TrySound) [reply .flowconfig:10]: `declarations` is like auto `$FlowFixMe` for matched modules. You still get some types but everything with errors becomes any.

File: src/index.js (hunk 2 above)
Comment (TrySound) [reply src/index.js:4]: Previously it was `{ $call: () => void }` but was recently changed to more explicit syntax.

Comment (TrySound) [reply src/index.js:4]: Function statics are more strict now and this is the only way to describe them properly.

File: .flowconfig (hunk 1 above)
Comment (alexreardon) [reply .flowconfig:10]: ah okay

**Pull request merged**